backend/src/main/java/com/capstone/moneytree/facade/StockMarketDataFacade.java

```diff
@@ -0,0 +1,136 @@
+import pl.zankowski.iextrading4j.client.IEXTradingApiVersion;
+import pl.zankowski.iextrading4j.client.IEXTradingClient;
+import pl.zankowski.iextrading4j.client.rest.request.stocks.*;
+import pl.zankowski.iextrading4j.client.rest.request.stocks.v1.BatchStocksRequestBuilder;
+import pl.zankowski.iextrading4j.client.rest.request.stocks.v1.BatchStocksType;
+import pl.zankowski.iextrading4j.client.rest.request.stocks.v1.KeyStatsRequestBuilder;
+import pl.zankowski.iextrading4j.client.rest.request.stocks.v1.NewsRequestBuilder;
+
+import java.util.List;
+
+/**
+ * This facade abstracts the IEXCloud API and exposes only the relevant
+ * methods for MoneyTree.
+ */
+@Component
+public class StockMarketDataFacade {
+
+   private final IEXCloudClient stockMarketDataClient;
+   private static final Logger LOG = LoggerFactory.getLogger(StockMarketDataFacade.class);
+
+   @Autowired
+   public StockMarketDataFacade(@Value("${IEXCloud.publishable.token}") String PUBLISH_TOKEN, @Value("${IEXCloud.secret.token}") String SECRET_TOKEN) {
+      stockMarketDataClient = IEXTradingClient.create(IEXTradingApiVersion.IEX_CLOUD_V1,
+              new IEXCloudTokenBuilder()
+                      .withPublishableToken(PUBLISH_TOKEN)
+                      .withSecretToken(SECRET_TOKEN)
+                .build());
+   }
+
+   public BatchStocks getBatchStocksBySymbol(String symbol) {
+      final BatchStocks batchStocks = stockMarketDataClient.executeRequest(new BatchStocksRequestBuilder()
+              .withSymbol(symbol)
+              .addType(BatchStocksType.BOOK)
+              .addType(BatchStocksType.COMPANY)
+              .addType(BatchStocksType.NEWS).withLast(10)
+              .addType(BatchStocksType.KEY_STATS)
+              .addType(BatchStocksType.LOGO)
+              .addType(BatchStocksType.CHART)
+              .build()
+      );
+
+      LOG.info("Fetch batch: {}", batchStocks.toString());
+      return batchStocks;
+   }
+
+   public Book getBook(String symbol) {
+      final Book book = stockMarketDataClient.executeRequest(new BookRequestBuilder()
+              .withSymbol(symbol)
+              .build());
+
+      LOG.info("Fetched book: {}", book.toString());
+      return book;
+   }
+
+   public Quote getQuote(String symbol) {
+      final Quote quote = stockMarketDataClient.executeRequest(new QuoteRequestBuilder()
+              .withSymbol(symbol)
+              .build());
+
+      LOG.info("Fetched quote: {}", quote.toString());
+      return quote;
+   }
+
+   public Company getCompanyInfo(String symbol) {
+      final Company company = stockMarketDataClient.executeRequest(new CompanyRequestBuilder()
+         .withSymbol(symbol)
+         .build());
+
+      LOG.info("Fetched company: {}", company.toString());
+      return company;
+   }
+
+   public List<News> getLastNNews(String symbol, int lastN) {
+      final List<News> news = stockMarketDataClient.executeRequest(new NewsRequestBuilder()
```

Comment:
Is there a limitations in lastN by the api? Could it be 2 or 50000?

Reply:
Yeah the API has some limitations that we'll have to manage. Either we mock the tests or we use the sandbox test API keys in dev env.